src/components/Popup/Popup.tsx

```diff
@@ -224,10 +262,94 @@ export default class Popup extends AutoControlledComponent<ReactProps<PopupProps
     )
   }
 
+  getTriggerProps(triggerElement) {
+    const triggerProps: any = {}
+
+    const { on } = this.props
+    const normalizedOn = _.isArray(on) ? on : [on]
+
+    if (_.includes(normalizedOn, 'click')) {
+      triggerProps.onClick = (e, ...rest) => {
+        this.trySetOpen(!this.state.open, e)
+        _.invoke(triggerElement, 'props.onClick', e, ...rest)
+      }
+    }
+    if (_.includes(normalizedOn, 'focus')) {
+      triggerProps.onFocus = (e, ...rest) => {
+        this.trySetOpen(true, e)
+        _.invoke(triggerElement, 'props.onFocus', e, ...rest)
+      }
+      triggerProps.onBlur = (e, ...rest) => {
+        if (
+          !e.currentTarget.contains(e.relatedTarget) &&
+          !this.popupDomElement.contains(e.relatedTarget)
+        ) {
+          this.trySetOpen(false, e)
+        }
+        _.invoke(triggerElement, 'props.onBlur', e, ...rest)
+      }
+    }
+    if (_.includes(normalizedOn, 'hover')) {
+      triggerProps.onMouseEnter = (e, ...rest) => {
+        this.setPopupOpen(true, e)
+        _.invoke(triggerElement, 'props.onMouseEnter', e, ...rest)
+      }
+      triggerProps.onMouseLeave = (e, ...rest) => {
+        this.setPopupOpen(false, e)
+        _.invoke(triggerElement, 'props.onMouseLeave', e, ...rest)
+      }
+      triggerProps.onClick = (e, ...rest) => {
+        this.setPopupOpen(true, e)
+        _.invoke(triggerElement, 'props.onClick', e, ...rest)
```

Comment:
From an accessibility perspective, shouldn't we open Popup which opens on 'hover', also on 'focus', not on 'click'?

Reply:
No, after talking with @jurokapsiar the decision was, the focus should not be default for Popups that have on hover (it can be optionally added with using `on={['focus', 'hover']}`). The hover Popups, should always be opened on click (this is not the toggling - just simple open on click) - so that the virtual cursors can first focus and then open the popup.